src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterThrottlingMetricsCollector.java

```diff
@@ -0,0 +1,112 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.ESResources;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsConfiguration;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsProcessor;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.elasticsearch.cluster.service.MasterService;
+
+import java.lang.reflect.Method;
+
+public class MasterThrottlingMetricsCollector extends PerformanceAnalyzerMetricsCollector implements MetricsProcessor {
+
+    public static final int SAMPLING_TIME_INTERVAL =
+            MetricsConfiguration.CONFIG_MAP.get(MasterThrottlingMetricsCollector.class).samplingInterval;
+    private static final Logger LOG = LogManager.getLogger(MasterThrottlingMetricsCollector.class);
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final String MASTER_THROTTLING_RETRY_LISTENER_PATH =
+            "org.elasticsearch.action.support.master.MasterThrottlingRetryListener";
+    private static final String THROTTLED_PENDING_TASK_COUNT_METHOD_NAME = "numberOfThrottledPendingTasks";
+    private static final String RETRYING_TASK_COUNT_METHOD_NAME = "getRetryingTasksCount";
+    private StringBuilder value;
+
+    public MasterThrottlingMetricsCollector() {
+        super(SAMPLING_TIME_INTERVAL, "MasterThrottlingMetricsCollector");
+        value = new StringBuilder();
+    }
+
+    @Override
+    void collectMetrics(long startTime) {
+        try {
+            if (ESResources.INSTANCE.getClusterService() == null
+                    || ESResources.INSTANCE.getClusterService().getMasterService() == null) {
```

Comment:
I am thinking we should add a metric here. You can add a metric like [so](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/master/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PublisherEventsPersistor.java#L47)

Reply:
We have added it at collector level, so it will be published for all collectors.

https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/436/files#diff-115bbe9019d03c22dc720eb9fbc6670f354808d16bebb1eb6933dd468ce470af

Reply:
Added metric as well.